src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFSDriver.Linux.cs

```diff
@@ -0,0 +1,562 @@
+        {
+            char buf;
+            IntPtr bufPtr = new IntPtr(&buf);
+            pinNumber = -1;
+
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                // poll every 1 millisecond
+                int waitResult = Interop.epoll_wait(pollFileDescriptor, out epoll_event events, 1, _pollingTimeoutInMilliseconds);
+                if (waitResult == -1)
+                {
+                    throw new IOException("Error while trying to initialize pin interrupts.");
+                }
+                if (waitResult > 0)
+                {
+                    pinNumber = events.data.pinNumber;
+
+                    // valueFileDescriptor will be -1 when using the callback eventing. For WaitForEvent, the value will be set.
+                    if (valueFileDescriptor == -1)
+                    {
+                        valueFileDescriptor = _devicePins[pinNumber].FileDescriptor;
+                    }
+
+                    int lseekResult = Interop.lseek(valueFileDescriptor, 0, SeekFlags.SEEK_SET);
+                    if (lseekResult == -1)
+                    {
+                        throw new IOException("Error while trying to initialize pin interrupts.");
+                    }
+
+                    int readResult = Interop.read(valueFileDescriptor, bufPtr, 1);
+                    if (readResult != 1)
+                    {
+                        throw new IOException("Error while trying to initialize pin interrupts.");
+                    }
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private void RemovePinFromPoll(int pinNumber, ref int valueFileDescriptor, ref int pollFileDescriptor, bool closePinValueFileDescriptor, bool closePollFileDescriptor, bool cancelEventDetectionThread)
+        {
+            epoll_event epollEvent = new epoll_event
+            {
+                events = PollEvents.EPOLLIN | PollEvents.EPOLLET | PollEvents.EPOLLPRI
+            };
+
+            int result = Interop.epoll_ctl(pollFileDescriptor, PollOperations.EPOLL_CTL_DEL, valueFileDescriptor, ref epollEvent);
+            if (result == -1)
+            {
+                throw new IOException("Error while trying to initialize pin interrupts.");
+            }
+
+            if (closePinValueFileDescriptor)
+            {
+                Interop.close(valueFileDescriptor);
+                valueFileDescriptor = -1;
+            }
+            if (closePollFileDescriptor)
+            {
+                if (cancelEventDetectionThread)
+                {
+                    s_eventThreadCancellationTokenSource.Cancel();
+                    while (_eventDetectionThread != null && _eventDetectionThread.IsAlive)
+                    {
+                        Thread.Sleep(TimeSpan.FromMilliseconds(10)); // Wait until the event detection thread is aborted.
```

Comment:
How do we know that 10 ms is sufficient?

Reply:
i don't really know how it calculated, anyways SysFs driver code didn't changed, just move to this new class as Unix driver now has 2 options, libgpiod and sysFs. I am gonna ignore sysFs related comments i dont really know or don't want to update